direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaAccessor.java

```diff
@@ -215,6 +232,44 @@ Properties createProps() {
         getUri(), createProps(), serializer.keySerde(), serializer.valueSerde());
   }
 
+  /**
+   * Checker for kafka topics configuration
+   *
+   * @param familyDescriptor Attribute family descriptor.
+   * @return true if check succeeded, otherwise false
+   */
+  @Override
+  public boolean isAcceptable(AttributeFamilyDescriptor familyDescriptor)
+      throws RuntimeException {
+    // Force checks for data compacting on state-commit-log topics
+    if (familyDescriptor.getAccess().isStateCommitLog()) {
+      try (AdminClient admin = createAdmin()) {
+        Collection<ConfigResource> cr =
+            Collections.singleton(new ConfigResource(ConfigResource.Type.TOPIC, this.topic));
+
+        DescribeConfigsResult configsResult = admin.describeConfigs(cr);
+
+        Config config = Iterables.getOnlyElement(configsResult.all().get().values());
+
+        for (ConfigEntry currentCfgEntry : config.entries()) {
+          if (currentCfgEntry.name().equals(TopicConfig.CLEANUP_POLICY_CONFIG)
+              && currentCfgEntry.value().contains(TopicConfig.CLEANUP_POLICY_COMPACT)) {
+            return true;
+          }
+        }
+
+        throw new IllegalStateException(
```

Comment:
We should be more explicit here about which topic has failed. eg:
`String.format('Missing option [cleanup.policy=compact] of kafka topic [%s] with access type [state-commit-log].', topic)`

Reply:
Hmm, we shouldn't throw an exception here. There could be other families for the ssame topic, that support this. I think whole method could be simplified as follows:

```java
  public boolean isAcceptable(AttributeFamilyDescriptor familyDescriptor)
      throws RuntimeException {
    // Force checks for data compacting on state-commit-log topics
    if (familyDescriptor.getAccess().isStateCommitLog()) {
      try (AdminClient admin = createAdmin()) {
        final DescribeConfigsResult configsResult = admin.describeConfigs(Collections.singleton(new ConfigResource(ConfigResource.Type.TOPIC, topic)));
        final Config config = Iterables.getOnlyElement(ExceptionUtils.uncheckedFactory(configsResult.all()::get).values());
        final ConfigEntry cleanupPolicy = config.get(TopicConfig.CLEANUP_POLICY_CONFIG);
        if (cleanupPolicy == null || !TopicConfig.CLEANUP_POLICY_COMPACT.equals(cleanupPolicy.value())) {
          log.warn("Missing option [cleanup.policy=compact] of kafka topic [{}] with access type [state-commit-log].", topic);
          return false;
        }
      }
    }
    return true;
  }
```

Reply:
Ah, that is correct! Good catch!